connection/go.mod

```diff
@@ -0,0 +1,36 @@
+module github.com/flipkart-incubator/go-dmux/connection
```

Comment:
Mod not required for every folder, instead lets move this to top level `github.com/flipkart-incubator/go-dmux` so that single